src/hotWaterHandler.h

```diff
@@ -17,7 +17,7 @@ void checkHotWaterSwitch() {
                 pumpRelay.on();
             }
 
-            if (hotWaterSwitchReading == LOW && machineState == kHotWater) {
+            if (hotWaterSwitchReading == LOW && hotWaterOn == 1) {
```

Comment:
if (hotWaterSwitchReading == LOW) {
 HotWater has its own machine state, which the machine is in as long as the toggle is high.
 No need to check the state of hotWaterOn